app/controllers/api/exchanges_products_controller.rb

```diff
@@ -1,6 +1,16 @@
 module Api
   class ExchangesProductsController < Api::BaseController
-    skip_authorization_check only: [:index]
+    skip_authorization_check only: [:index, :show]
+
+    def show
```

Comment:
ExchangeProductsController is about products that CAN be added to an exchange. I'll add this to the controllers documentation. I dont think this can be mixed up with exchange controller that handles data in the exchange. Even for handling adding variants to an exchange I'd probably go for a new controller called ExchangeVariants.

I think that after I extracted the ExchangeProductsRenderer this ExchangeProductsController became nice and small just handling the different combinations of the params given.

The only major standards/consistency flaw here is that I cant make 
api/exchanges/products (without exchange id) redirect to the ExchangeProductsController#index, it goes to ExchangeController#show and manage to redirect it to ExchangeProductsController#show which is better but not nice...

This controller resource is an product in an exchange. So the show action should

Reply:
ah, I managed to make api/exchanges/products go to "exchange_products#index" 🎉 
e177dc5cd5b0c91481c00a95e04640150987b056

Reply:
ah, I managed to make api/exchanges/products go to "exchange_products#index" 🎉
e177dc5

Reply:
Hm, I think I preferred to have two separate methods instead of if-else. It means that Rails is doing the if-else for you. But it's not that important. I'm happy for you to leave it as is.